### ISSUE TITLE
Functions called with deeply equal arguments (and deepEqual=true) seemingly not being memoized

### ISSUE THREAD

**Issue by tnrich** [erikras/lru-memoize#73]
Hey there @erikras,

Thanks for the great tool. I'm having an issue where calling my memoized function with deeply nested objects that are deeply equal and having the deepEqual option set to true still results in a new function call.

Here's an example of this happening:

```
var lruMemoize = require("lru-memoize")
function test(options) {
    console.log('HELLO WORLD');
    return 'hello'
}

var mTest = lruMemoize(1,undefined,true)(test)

mTest({
    a: {
        yup: 'nope'
    },
    b: {
        well: 'no'
    }
})
mTest({
    a: {
        yup: 'nope'
    },
    b: {
        well: 'no'
    }
})
```

In this case "Hello World" is output to the console twice, instead of just once.

Any help solving this would be appreciated. Thanks!
Thomas

**Comment by oculus42** [comment]
The way the config is handled, it doesn't handle the `undefined` value for the equal function. If you remove the undefined, this will work. It would be nice if it supported the undefined, as well.

**Comment by erikras** [comment]
Published fix in [`v1.0.2`](https://github.com/erikras/lru-memoize/releases/tag/v1.0.2).
